### PR TITLE
update CSI driver, enable multi-node

### LINKED ISSUE #12360: Persistent Volumes permission issue for 2+ node clutsters
Hello,
If minikube cluster is 2+ nodes then initContainers cannot set volume permissions inside PVCs and pods fail to access data there. Single node cluster do not have this issue.

### PR DESCRIPTION
**Pull request by denisok**
fixes #12360

### CODE REVIEW

File: deploy/addons/csi-hostpath-driver/deploy/csi-hostpath-plugin.yaml.tmpl [hunk 1]
```diff
@@ -16,64 +16,37 @@
...
+        app.kubernetes.io/part-of: csi-driver-host-path
+        app.kubernetes.io/name: csi-hostpathplugin
+        app.kubernetes.io/component: plugin
         addonmanager.kubernetes.io/mode: Reconcile
         kubernetes.io/minikube-addons: csi-hostpath-driver
     spec:
-      serviceAccount: csi-external-health-monitor-controller
+      serviceAccount: csi-hostpathplugin-sa
       containers:
-        - name: csi-external-health-monitor-agent
```
Comment (denisok) [inline deploy/addons/csi-hostpath-driver/deploy/csi-hostpath-plugin.yaml.tmpl:61]: health-agent was merged to kubelet: https://github.com/kubernetes-csi/external-health-monitor#overview

File: deploy/addons/csi-hostpath-driver/rbac/rbac-external-provisioner.yaml.tmpl [hunk 2]
```diff
@@ -92,12 +96,24 @@ metadata:
 rules:
 # Only one of the following rules for endpoints or leases is required based on
 # what is set for `--leader-election-type`. Endpoints are deprecated in favor of Leases.
-- apiGroups: [""]
```
Comment (denisok) [inline deploy/addons/csi-hostpath-driver/rbac/rbac-external-provisioner.yaml.tmpl:95]: https://github.com/kubernetes-csi/external-provisioner/blob/v3.3.0/deploy/kubernetes/rbac.yaml

**Review by medyagh — COMMENTED** [review]
do you mind adding output of minikube Before / After this PR?

also, it would be nice to add a section on multinode use example in our website

https://minikube.sigs.k8s.io/docs/tutorials/volume_snapshots_and_csi/

**Comment by denisok** [comment]
> do you mind adding output of minikube Before / After this PR?

sure, here evertyhing with `./out/` is a new version:

```sh
[dkondratenko@denlen minikube]$ minikube addons enable volumesnapshots 
💡  volumesnapshots is an addon maintained by Kubernetes. For any concerns contact minikube on GitHub.
You can view the list of minikube maintainers at: https://github.com/kubernetes/minikube/blob/master/OWNERS
    ▪ Using image k8s.gcr.io/sig-storage/snapshot-controller:v4.0.0
🌟  The 'volumesnapshots' addon is enabled
[dkondratenko@denlen minikube]$ minikube addons disable volumesnapshots 
🌑  "The 'volumesnapshots' addon is disabled

[dkondratenko@denlen minikube]$ ./out/minikube addons enable volumesnapshots 
💡  volumesnapshots is an addon maintained by Kubernetes. For any concerns contact minikube on GitHub.
You can view the list of minikube maintainers at: https://github.com/kubernetes/minikube/blob/master/OWNERS
    ▪ Using image registry.k8s.io/sig-storage/snapshot-controller:v6.1.0
🌟  The 'volumesnapshots' addon is enabled
[dkondratenko@denlen minikube]$ ./out/minikube addons disable volumesnapshots 
🌑  "The 'volumesnapshots' addon is disabled

[dkondratenko@denlen minikube]$ minikube addons enable csi-hostpath-driver
💡  csi-hostpath-driver is an addon maintained by Kubernetes. For any concerns contact minikube on GitHub.
You can view the list of minikube maintainers at: https://github.com/kubernetes/minikube/blob/master/OWNERS
❗  [WARNING] For full functionality, the 'csi-hostpath-driver' addon requires the 'volumesnapshots' addon to be enabled.

You can enable 'volumesnapshots' addon by running: 'minikube addons enable volumesnapshots'

    ▪ Using image k8s.gcr.io/sig-storage/csi-snapshotter:v4.0.0
    ▪ Using image k8s.gcr.io/sig-storage/csi-node-driver-registrar:v2.0.1
    ▪ Using image k8s.gcr.io/sig-storage/livenessprobe:v2.2.0
    ▪ Using image k8s.gcr.io/sig-storage/csi-provisioner:v2.1.0
    ▪ Using image k8s.gcr.io/sig-storage/csi-external-health-monitor-controller:v0.2.0
    ▪ Using image k8s.gcr.io/sig-storage/hostpathplugin:v1.6.0
    ▪ Using image k8s.gcr.io/sig-storage/csi-resizer:v1.1.0
    ▪ Using image k8s.gcr.io/sig-storage/csi-attacher:v3.1.0
    ▪ Using image k8s.gcr.io/sig-storage/csi-external-health-monitor-agent:v0.2.0
🔎  Verifying csi-hostpath-driver addon...
🌟  The 'csi-hostpath-driver' addon is enabled
[dkondratenko@denlen minikube]$ minikube addons disable csi-hostpath-driver
🌑  "The 'csi-hostpath-driver' addon is disabled

[dkondratenko@denlen minikube]$ ./out/minikube addons enable csi-hostpath-driver
💡  csi-hostpath-driver is an addon maintained by Kubernetes. For any concerns contact minikube on GitHub.
You can view the list of minikube maintainers at: https://github.com/kubernetes/minikube/blob/master/OWNERS
❗  [WARNING] For full functionality, the 'csi-hostpath-driver' addon requires the 'volumesnapshots' addon to be enabled.

You can enable 'volumesnapshots' addon by running: 'minikube addons enable volumesnapshots'

    ▪ Using image registry.k8s.io/sig-storage/csi-node-driver-registrar:v2.6.0
    ▪ Using image registry.k8s.io/sig-storage/hostpathplugin:v1.9.0
    ▪ Using image registry.k8s.io/sig-storage/livenessprobe:v2.8.0
    ▪ Using image registry.k8s.io/sig-storage/csi-resizer:v1.6.0
    ▪ Using image registry.k8s.io/sig-storage/csi-snapshotter:v6.1.0
    ▪ Using image registry.k8s.io/sig-storage/csi-provisioner:v3.3.0
    ▪ Using image registry.k8s.io/sig-storage/csi-attacher:v4.0.0
    ▪ Using image registry.k8s.io/sig-storage/csi-external-health-monitor-controller:v0.7.0
🔎  Verifying csi-hostpath-driver addon...
🌟  The 'csi-hostpath-driver' addon is enabled
[dkondratenko@denlen minikube]$ ./out/minikube addons disable csi-hostpath-driver
🌑  "The 'csi-hostpath-driver' addon is disabled
```

> also, it would be nice to add a section on multinode use example in our website
> 
> https://minikube.sigs.k8s.io/docs/tutorials/volume_snapshots_and_csi/
yes, will do later.

**Comment by denisok** [comment]
I see some failures related to csi, will check it.

**Comment by denisok** [comment]
OK, I looked to some of the tests, and those failures related to `current state: Pending` for many containers not just csi drivers. Same in other PRs currently open.

I tried different confirurations and don't see any issues with kvm or podman driver,

Once I seen the issue minikube starting with podman and cri-o - Node wasn't ready and thus containers were in `Pending` state, looks like #15676 .

```
Conditions:
  Type             Status  LastHeartbeatTime                 LastTransitionTime                Reason                       Message
  ----             ------  -----------------                 ------------------                ------                       -------
  MemoryPressure   False   Mon, 13 Feb 2023 19:39:17 +0100   Mon, 13 Feb 2023 19:39:12 +0100   KubeletHasSufficientMemory   kubelet has sufficient memory available
  DiskPressure     False   Mon, 13 Feb 2023 19:39:17 +0100   Mon, 13 Feb 2023 19:39:12 +0100   KubeletHasNoDiskPressure     kubelet has no disk pressure
  PIDPressure      False   Mon, 13 Feb 2023 19:39:17 +0100   Mon, 13 Feb 2023 19:39:12 +0100   KubeletHasSufficientPID      kubelet has sufficient PID available
  Ready            False   Mon, 13 Feb 2023 19:39:17 +0100   Mon, 13 Feb 2023 19:39:12 +0100   KubeletNotReady              container runtime network not ready: NetworkReady=false reason:NetworkPluginNotReady message:Network plugin returns error: No CNI configuration file in /etc/cni/net.d/. Has your network provider started?
```

But it was only once and I can't reproduce it.

`PR / functional_podman_ubuntu (pull_request)` fails due to network issue with GH or network:
```
10s
Run curl -LO https://github.com/medyagh/gopogh/releases/download/v0.13.0/gopogh-linux-amd64
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0

  0 3848k    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
 73 3848k   73 2831k    0     0  1985k      0  0:00:01  0:00:01 --:--:-- 2686k
 73 3848k   73 2831k    0     0  1166k      0  0:00:03  0:00:02  0:00:01 1377k
 73 3848k   73 2831k    0     0   825k      0  0:00:04  0:00:03  0:00:01  926k
 73 3848k   73 2831k    0     0   639k      0  0:00:06  0:00:04  0:00:02  697k
 73 3848k   73 2831k    0     0   521k      0  0:00:07  0:00:05  0:00:02  559k
 73 3848k   73 2831k    0     0   440k      0  0:00:08  0:00:06  0:00:02     0
 73 3848k   73 2831k    0     0   380k      0  0:00:10  0:00:07  0:00:03     0
 73 3848k   73 2831k    0     0   335k      0  0:00:11  0:00:08  0:00:03     0
 73 3848k   73 2831k    0     0   300k      0  0:00:12  0:00:09  0:00:03     0
 73 3848k   73 2831k    0     0   272k      0  0:00:14  0:00:10  0:00:04     0
 73 3848k   73 2831k    0     0   272k      0  0:00:14  0:00:10  0:00:04     0
curl: (18) transfer closed with 1041185 bytes remaining to read
Error: Process completed with exit code 18.
```

needs re-run.

**Comment by denisok** [comment]
Hi @medyagh,

I added docs and checked tests. So far I found no regressions.

Thanks,
Denys

**Review by medyagh — APPROVED** [review]
Thank you for this PR, please consider making a follow UP PR that when users use multi-node, minikube automatically enable this addon for them

**Pull request merged**